ceph/rados/serviceability_workflows.py

```diff
@@ -315,6 +315,87 @@ def remove_osds_from_host(self, host_obj):
                     service_type="osd", service_name=service
                 )
 
+    def add_osds_to_managed_service(self, osds: list = None, spec: str = None) -> bool:
+        """
+        Method to identify non-managed OSDs and then add them back to managed state under existing OSD specs
+        Args:
+            osds: list of OSDs to be set to managed state
+            spec: name of the spec that needs to be used for setting OSD to manage.
+        Returns:
+            Pass -> True
+            fail -> False
+
+        Note : if osds list is provided, Only those osds will be added to managed and check if the cluster,
+        is free of unmanaged OSDs.
+
+        Note : If no OSDs are provided, we add all OSDs to managed host by host,
+        until we're left with no unmanaged osds on cluster.
+
+        Note: If Spec is not provided, 1st available - managed OSD spec name would be used
+        """
+
+        def get_osd_services():
+            cmd_export = "ceph orch ls osd --export"
+            return self.rados_obj.run_ceph_command(cmd=cmd_export, client_exec=True)
```

Comment:
We have a method for this in core_workflows.

`out = rados_obj.list_orch_services(service_type="osd", export=True)`

Reply:
Updated method call